packages/reflection/src/ClassReflector.php

```diff
@@ -167,4 +169,13 @@ public function implements(string $interface): bool
     {
         return $this->isInstantiable() && $this->getType()->matches($interface);
     }
+
+    private function memoize(string $key, Closure $closure): mixed
+    {
+        if (! isset($this->memoize[$key])) {
```

Comment:
Should we use `array_key_exists`  to not call the `$closure` again in case it returns null?

Reply:
Ah, yes, good point 😅

Reply:
https://github.com/tempestphp/tempest-framework/pull/1360 :)